lib/plugins/physics.js

```diff
@@ -78,35 +98,54 @@ function inject (bot, { physicsEnabled }) {
   function sendPacketPosition (position, onGround) {
     // sends data, no logic
     const oldPos = new Vec3(lastSent.x, lastSent.y, lastSent.z)
-    lastSent.x = position.x
-    lastSent.y = position.y
-    lastSent.z = position.z
-    lastSent.onGround = onGround
-    bot._client.write('position', lastSent)
+    const packet = {
+      x: position.x,
+      y: position.y,
+      z: position.z,
+      onGround: onGround
+    }
+    packet.onGround = onGround
```

Comment:
You are assigning onGround twice. And you are removing an optimization: lastSent was used to avoid creating too many temporary objects.

Reply:
That was a my bad on the packet.onGround part.
I have changed it to use lastSent.
However lastSent.onGround is always updated in vanilla and lastSent.time is always reset when a position or a position_look packet is sent.